src/app/public/src/modules/edit-button/edit-button.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+import { StacheJsonDataService, StacheRouteService } from '../shared';
+import { StacheNavService } from '../nav';
+
+@Component({
+  selector: 'stache-edit-button',
+  templateUrl: './edit-button.component.html'
+})
+export class StacheEditButtonComponent implements OnInit {
+  @Input()
+  public editButtonText = 'Edit';
+
+  private url: string;
+
+  public constructor (
+    private dataService: StacheJsonDataService,
+    private navService: StacheNavService,
+    private routeService: StacheRouteService
+  ) { }
+
+  public ngOnInit() {
+    this.url = this.getUrl();
+  }
+
+  public onEditButtonClick() {
+    this.navService.navigate({ path: this.url });
+  }
+
+  private getUrl(): string {
+    const repoData = this.dataService.getByName('repo');
+    const frag = encodeURIComponent(this.routeService.getActiveUrl() + '/');
+    const result = repoData.url + frag + 'index.html';
+
+    return (repoData.vstsBranchSelector === undefined) ? result : result + repoData.vstsBranchSelector;
+  }
+}
```

Comment:
I can't add a comment to the `edit-button.module.spec.ts` file below, but it appears to be empty.